### PR TITLE
Add possiblity to sort SGNs in GUI based on numerical value

### PR DESCRIPTION
**Pull request by WeirdRubberDuck**
Adds property the scene graph nodes for sorting the item in the GUI based on a number instead of alphabetically, with respect to its neighboring scene graph nodes in the menu. 

Let me know what you think of the property name `OrderNumber`. Didn't come up with anything better...

This PR should be combined with the following PR in the WebGUI repo: https://github.com/OpenSpace/OpenSpace-WebGuiFrontend/pull/192

Also applies the new sorting to the grids, to sort them based on their size instead of alphabetically:

image

Note that: 
- Another PR is coming for also controlling the order of the _Groups_ in the UI.
- We do not care about using this ordering in the ImGui UI. Hence why it is not applied there

### CODE REVIEW

**Comment by alexanderbock** [comment]
Could go to "Order" order "Ordering" instead?

File: include/openspace/scene/scenegraphnode.h [hunk 1]
```diff
@@ -170,15 +172,17 @@ class SceneGraphNode : public properties::PropertyOwner {
     std::vector<std::string> _onRecedeAction;
     std::vector<std::string> _onExitAction;
 
+    ghoul::mm_unique_ptr<Renderable> _renderable;
+
```
Comment (alexanderbock) [inline include/openspace/scene/scenegraphnode.h:176]: Why moving it further up?

File: src/scene/scenegraphnode.cpp [hunk 2]
```diff
@@ -309,6 +326,15 @@ namespace {
...
             // not display, for example, barycenters
             std::optional<bool> hidden;
+
+            // If this value is specified, the scene graph node will be ordered in
+            // relation to its neighbors in the GUI based on this value, so that nodes
+            // with a higher value appear later in the list. Scene graph noeds with the
+            // same value will be sorted alphabetically.
+            //
+            // The nodes without a given value will be placed at the top at the list and
+            // sorted alphabetically.
```
Comment (alexanderbock) [inline src/scene/scenegraphnode.cpp:336]: I thought we wanted to move the items without to the bottom of the list?

**Review by alexanderbock — APPROVED** [review]

File: include/openspace/scene/scenegraphnode.h (hunk 1 above)
Comment (WeirdRubberDuck) [reply include/openspace/scene/scenegraphnode.h:176]: Ah, just to group the gui properties. It bugged me that `_guiHidden` was not placed together with the other ones

But I'm fine with undoing this change

File: src/scene/scenegraphnode.cpp (hunk 2 above)
Comment (WeirdRubberDuck) [reply src/scene/scenegraphnode.cpp:336]: I did, just forgot to change this comment  :) Thanks

**Comment by WeirdRubberDuck** [comment]
> Could go to "Order" order "Ordering" instead?

Hm, yeah, maybe. Sound a bit less like the kind of number you get when making an online purchase at least 😅

File: include/openspace/scene/scenegraphnode.h (hunk 1 above)
Comment (alexanderbock) [reply include/openspace/scene/scenegraphnode.h:176]: Nah.. i was just curious

File: src/scene/scenegraphnode.cpp [hunk 3]
```diff
@@ -175,6 +175,23 @@ namespace {
         openspace::properties::Property::Visibility::Hidden
     };
 
+    constexpr openspace::properties::Property::PropertyInfo GuiOrderInfo = {
+        "GuiOrderNumber",
+        "Gui Order Number",
+        "This is an optional numerical value that will affect the sorting of this scene "
+        "graph node in relation ot its neighbors in the user interface. Nodes with the "
```
Comment (ylvaselling) [inline src/scene/scenegraphnode.cpp:182]:
```suggestion
        "graph node in relation to its neighbors in the GUI. Nodes with the "
```

File: src/scene/scenegraphnode.cpp [hunk 4]
```diff
@@ -175,6 +175,23 @@ namespace {
...
+        "same value will be sorted alphabetically.",
+        openspace::properties::Property::Visibility::Hidden
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo UseGuiOrderInfo = {
+        "UseGuiOrder",
+        "UseGuiOrder",
```
Comment (ylvaselling) [inline src/scene/scenegraphnode.cpp:189]:
```suggestion
        "Use Gui Order",
```

File: src/scene/scenegraphnode.cpp [hunk 5]
```diff
@@ -309,6 +326,15 @@ namespace {
             // scene graph node. This is most useful to trim collective lists of nodes and
             // not display, for example, barycenters
             std::optional<bool> hidden;
+
+            // If this value is specified, the scene graph node will be ordered in
+            // relation to its neighbors in the GUI based on this value, so that nodes
+            // with a higher value appear later in the list. Scene graph noeds with the
```
Comment (ylvaselling) [inline src/scene/scenegraphnode.cpp:332]:
```suggestion
            // with a higher value appear later in the list. Scene graph nodes with the
```

**Review by ylvaselling — CHANGES_REQUESTED** [review]

**Pull request merged**